src/App.jsx

```diff
@@ -14,45 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FILED_ALPHABETICALLY = 'alphabetically';
+const SORT_FILED_BY_LENGTH = 'length';
+const REVERSE = true;
+
+function getPreparedGoods(goods, { sortField, reverse }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((firstGood, secondGood) => {
+      switch (sortField) {
+        case SORT_FILED_ALPHABETICALLY:
+          return firstGood.localeCompare(secondGood);
+
+        case SORT_FILED_BY_LENGTH:
+          return firstGood.length - secondGood.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse === REVERSE) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const goodsCopy = getPreparedGoods(goodsFromServer, { sortField, reverse });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FILED_ALPHABETICALLY,
+          })}
+          onClick={() => {
+            setSortField(SORT_FILED_ALPHABETICALLY);
+          }}
```

Comment:
It's better to create separate functions to handle events. This way, your code will be easier to maintain and change
```suggestion
          onClick={onSortAlphabetically}
```